pkg/etcdutil/etcdutil.go

```diff
@@ -248,25 +248,31 @@ func GetEtcdEndPointsSorted(ctx context.Context, clientMaintenance client.Mainte
 
 // TakeAndSaveFullSnapshot takes full snapshot and save it to store
 func TakeAndSaveFullSnapshot(ctx context.Context, client client.MaintenanceCloser, store brtypes.SnapStore, lastRevision int64, cc *compressor.CompressionConfig, suffix string, isFinal bool, logger *logrus.Entry) (*brtypes.Snapshot, error) {
+	startTime := time.Now()
 	rc, err := client.Snapshot(ctx)
 	if err != nil {
 		return nil, &errors.EtcdError{
 			Message: fmt.Sprintf("failed to create etcd snapshot: %v", err),
 		}
 	}
+	timeTaken := time.Since(startTime)
+	logger.Infof("Total time taken by Snapshot API: %f seconds.", timeTaken.Seconds())
+
 	if cc.Enabled {
+		startTimeCompression := time.Now()
 		rc, err = compressor.CompressSnapshot(rc, cc.CompressionPolicy)
 		if err != nil {
 			return nil, fmt.Errorf("unable to obtain reader for compressed file: %v", err)
 		}
+		timeTakenCompression := time.Since(startTimeCompression)
+		logger.Infof("Total time taken in full snapshot compression: %f seconds.", timeTakenCompression.Seconds())
 	}
 	defer rc.Close()
 
 	logger.Infof("Successfully opened snapshot reader on etcd")
 
 	// Then save the snapshot to the store.
 	snapshot := snapstore.NewSnapshot(brtypes.SnapshotKindFull, 0, lastRevision, suffix, isFinal)
-	startTime := time.Now()
```

Comment:
Do we not want any logs for how long a snapshot upload took?

Reply:
hmm ...basically this can deduce from above information.
we have `timeTaken` which is timeTaken in api call + compression time + upload time.
we log the  Snapshot API call time, log compress time, log the total time taken. I think we this information we can Calculate the snapshot upload timee.

Reply:
Right, makes sense 
Thanks